presto-resource-group-managers/src/test/java/com/facebook/presto/resourceGroups/TestFileResourceGroupConfigurationManager.java

```diff
@@ -51,15 +53,49 @@ public void testMissing()
     {
         ResourceGroupConfigurationManager manager = parse("resource_groups_config.json");
         ResourceGroup missing = new TestingResourceGroup(new ResourceGroupId("missing"));
-        manager.configure(missing, new SelectionContext(true, "user", Optional.empty(), 1));
+        manager.configure(missing, new SelectionContext(true, "user", Optional.empty(), 1, false));
+    }
+
+    @Test
+    public void testMetadataConfiguration()
+    {
+        ResourceGroupConfigurationManager manager = parse("resource_groups_config_metadata.json");
+        List<ResourceGroupSelector> selectors =  manager.getSelectors();
+        assertTrue(selectors.size() == 2, "there should be two selectors");
+        ResourceGroupSelector dataOnly = selectors.get(0);
+        ResourceGroupSelector withMetadata = selectors.get(1);
+
+        //data query against the first selector
+        SelectionContext context = new SelectionContext(true, "test_user_1", Optional.empty(), 1, false);
+        Optional<ResourceGroupId> group = dataOnly.match(context);
+        assertTrue(group.isPresent(), "match expected");
+        assertTrue(group.get().toString().equals("global.data"), "unexpected match");
```

Comment:
Can we do `assertEquals(group.get().toString(), "global.data")` instead? In addition, "unexpected match" does seem to be a useful message.